service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/UpdateGcsBucketStepTest.java

```diff
@@ -105,7 +105,9 @@ public void testDoStep() throws InterruptedException, RetryException {
 
     final LifecycleCondition rule1condition = rules.get(0).getCondition();
     assertEquals(31, rule1condition.getAge());
-    assertEquals(toGoogleDateTime(OFFSET_DATE_TIME_2), rule1condition.getCreatedBefore());
```

Comment:
This is the issue I discussed offline: Jackson serialization serializes `OffsetDateTime` as a double-precision UTC epoch, effectively dropping the time zone info.  Jackson can be configured to keep time zone info, but it changes the representation from a double to a string, so is completely non-upward-compatible.

Note that this test change normalizes both time stamps to a UTC epoch (moral equivalent to `Instant`) and compares.  If we need to keep time zone on-disk we will need to do something special.

Reply:
FWIW, I still think it's incorrect to put OffsetDateTime in a a flightmap if it's silently dropping the timezone info, but maybe I've missed more follow up discussion.
I would rather us do explicit, custom serialization of the offending class, or use a type that can be serdes'd correctly by jackson. It's kind of surprising to me that jackson doesn't really support java8 time classes.

Reply:
I agree that the OffsetDateTime is incorrect; the test doesn't matter so much, but the production code should be fixed.

Jackson does support Java8 time classes. John mentioned this in another communication. To get it to work, we would need to enable WRITE_DATES_AS_TIMESTAMPS. The risk of adding that is compatibility with existing serialized data. 

I added PF-933 to track the work to fix it, leaving it open whether to hand-fix the OffsetDateTime or reviewing existing Stairway users to see if there would be compatibility issues with enabling WRITE_DATES_AS_TIMESTAMPS.

Reply:
If the change to use `.value` in these tests is what I think it is, consider adding `TODO(PF-933)`'s to change the test from asserting that just the value is equal back to the entire offsetdatetime.